kmk/kmk_keyboard.py

```diff
@@ -458,7 +460,9 @@ def _main_loop(self):
         for matrix in self.matrix:
             update = matrix.scan_for_changes()
             if update:
-                self.matrix_update = update
+                self.matrix_update = KeyEvent(
+                    key_number=update.key_number, pressed=update.pressed
+                )
```

Comment:
If this is caused by re-using a single KeyEvent object, it should be fixed in `scanners.keypad`.

Reply:
Thanks, that makes sense.  I moved the changes to keypad.py

Reply:
In that case you can remove `KeypadScanner.current_event` completely and use `keypad.events.get` instead of `get_into`.

Reply:
In that case you can remove `KeypadScanner.current_event` completely and use `keypad.events.get` instead of `get_into`.

Reply:
I removed `KeypadScanner.curr_event` and made the function work without it.  I also removed `KeypadScanner.__init__()` as it no longer contains any code.